### ISSUE TITLE
Failed upgrade from 12.0.4 → 13.0.0.6

### ISSUE THREAD

**Issue by 9662** [nextcloud/server#7439]
### Steps to reproduce
1. Upgraded via web interface from 12.0.4 to 13.0.0.6 (13.0.0 Beta 1)
2. On install finished, clicked on "switch maintenance mode off" option to continue with web upgrade

### Expected behaviour
Upgrade should complete and user gets taken back to Nextcloud UI

### Actual behaviour
HTTP 503. The server log shows:
```
PHP Fatal error:  Declaration of OCA\\FilesAccessControl\\StorageWrapper::copyFromStorage(OCP\\Files\\Storage $sourceStorage, $sourceInternalPath, $targetInternalPath) must be compatible with OCP\\Files\\Storage::copyFromStorage(OCP\\Files\\Storage\\IStorage $sourceStorage, $sourceInternalPath, $targetInternalPath) in /<…trimmed…>/nextcloud/apps/files_accesscontrol/lib/StorageWrapper.php on line 628
```

### Server configuration

**Operating system**: openSUSE 42.2

**Web server:** Apache 2.4.23

**Database:** SQLite3

**PHP version:** 7.1.6

**Nextcloud version:** 13.0.0.6

**Updated from an older Nextcloud/ownCloud or fresh install:** Up from 12.0.4

**Where did you install Nextcloud from:** Nextcloud.com

**Signing status:**
<details>
<summary>Signing status</summary>

```
Unable to login
```
</details>

**List of activated apps:**
<details>
<summary>App list</summary>

**Note PHP error when running command**

```
./occ app:list
PHP Warning:  PHP Startup: Invalid library (appears to be a Zend Extension, try loading using zend_extension=opcache.so from php.ini) in Unknown on line 0
Nextcloud or one of the apps require upgrade - only a limited number of commands are available
You may use your browser or the occ upgrade command to do the upgrade
Enabled:
  - activity: 2.5.2
  - admin_audit: 1.2.0
  - bookmarks: 0.10.1
  - bruteforcesettings: 1.0.2
  - calendar: 1.5.6
  - comments: 1.2.0
  - contacts: 2.0.1
  - dav: 1.3.0
  - deck: 0.2.8
  - drawio: 0.8.8
  - federatedfilesharing: 1.2.0
  - federation: 1.2.0
  - files: 1.7.2
  - files_accesscontrol: 1.2.5
  - files_automatedtagging: 1.2.2
  - files_downloadactivity: 1.1.1
  - files_external: 1.3.0
  - files_markdown: 2.0.1
  - files_pdfviewer: 1.1.1
  - files_retention: 1.1.2
  - files_sharing: 1.4.0
  - files_texteditor: 2.4.1
  - files_trashbin: 1.2.0
  - files_versions: 1.5.0
  - files_videoplayer: 1.1.0
  - firstrunwizard: 2.1
  - gallery: 17.0.0
  - gpxpod: 2.2.1
  - logreader: 2.0.0
  - lookup_server_connector: 1.0.0
  - metadata: 0.6.0
  - nextcloud_announcements: 1.1
  - notes: 2.3.1
  - notifications: 2.0.0
  - oauth2: 1.0.5
  - ocsms: 1.12.1
  - password_policy: 1.2.2
  - polls: 0.8.0
  - provisioning_api: 1.2.0
  - quota_warning: 1.1.1
  - serverinfo: 1.2.0
  - sharebymail: 1.2.0
  - spreed: 2.0.2
  - survey_client: 1.0.0
  - systemtags: 1.2.0
  - tasks: 0.9.5
  - telephoneprovider: 1.0.0
  - theming: 1.3.0
  - twofactor_backupcodes: 1.1.1
  - twofactor_totp: 1.3.1
  - updatenotification: 1.2.0
  - user_external: 0.4
  - workflowengine: 1.2.0
Disabled:
  - encryption
  - user_ldap
```
</details>

**Nextcloud configuration:**
<details>
<summary>Config report</summary>

```
Nextcloud or one of the apps require upgrade - only a limited number of commands are available
You may use your browser or the occ upgrade command to do the upgrade
{
    "system": {
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "***REMOVED SENSITIVE VALUE***"
        ],
        "auth.bruteforce.protection.enabled": false,
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "overwrite.cli.url": "https:\/\/***REMOVED SENSITIVE VALUE***\/oc",
        "dbtype": "sqlite3",
        "version": "12.0.4.1",
        "installed": true,
        "loglevel": 1,
        "theme": "",
        "maintenance": false,
        "share_folder": "\/Shared",
        "mail_smtpmode": "smtp",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpauthtype": "LOGIN",
        "mail_smtpauth": true,
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpsecure": "ssl",
        "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
        "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "465",
        "forcessl": true,
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trashbin_retention_obligation": "auto",
        "updatechecker": false,
        "appstore.experimental.enabled": false,
        "updater.release.channel": "beta",
        "updater.secret": "***REMOVED SENSITIVE VALUE***"
    },
    "apps": {
        "activity": {
            "installed_version": "2.5.2",
            "types": "filesystem",
            "enabled": "yes"
        },
        "admin_audit": {
            "installed_version": "1.2.0",
            "types": "logging",
            "enabled": "yes"
        },
        "backgroundjob": {
            "lastjob": "1160"
        },
        "bookmarks": {
            "installed_version": "0.10.1",
            "types": "",
            "enabled": "yes"
        },
        "bruteforcesettings": {
            "installed_version": "1.0.2",
            "enabled": "yes",
            "types": ""
        },
        "calendar": {
            "installed_version": "1.5.6",
            "types": "",
            "enabled": "yes",
            "signed": "true"
        },
        "comments": {
            "installed_version": "1.2.0",
            "types": "logging",
            "enabled": "yes"
        },
        "contacts": {
            "installed_version": "2.0.1",
            "types": "",
            "enabled": "yes",
            "ocsid": "168708"
        },
        "core": {
            "installedat": "1397392290.6372",
            "lastupdatedat": "1512910239",
            "remote_core.css": "\/core\/minimizer.php",
            "remote_core.js": "\/core\/minimizer.php",
            "public_files": "files_sharing\/public.php",
            "public_webdav": "dav\/appinfo\/v1\/publicwebdav.php",
            "remote_calendar": "dav\/appinfo\/v1\/caldav.php",
            "remote_caldav": "dav\/appinfo\/v1\/caldav.php",
            "public_calendar": "calendar\/share.php",
            "public_caldav": "calendar\/share.php",
            "remote_contacts": "dav\/appinfo\/v1\/carddav.php",
            "remote_carddav": "dav\/appinfo\/v1\/carddav.php",
            "public_gallery": "gallery\/public.php",
            "remote_files": "dav\/appinfo\/v1\/webdav.php",
            "remote_webdav": "dav\/appinfo\/v1\/webdav.php",
            "remote_filesync": "files\/appinfo\/filesync.php",
            "public_documents": "documents\/public.php",
            "global_cache_gc_lastrun": "1442016702",
            "lastupdateResult": "{\"version\":\"13.0.0.6\",\"versionstring\":\"Nextcloud 13.0.0 beta 1\",\"url\":\"https:\\\/\\\/download.nextcloud.com\\\/server\\\/prereleases\\\/nextcloud-13.0.0beta1.zip\",\"web\":\"https:\\\/\\\/docs.nextcloud.com\\\/server\\\/12\\\/admin_manual\\\/maintenance\\\/upgrade.html\",\"autoupdater\":\"1\",\"signature\":\"TJVgSXFSR4ZG2eKzcSVbHjfxcJv4ieXI0uEUgtwcoAwma0oLm88OICkNX79c9YlA\\nmyF9LD5MxUFpWAi2dxOy4CGvXdWqx6o5DoCDGODNzDgwKpOqFd4xl2JqLcs4mYcz\\n1XkA61Mr7RkfiUPc6Hw+\\\/Q851vhD7G+4KvS4bj6m6UDaYrpDw\\\/9mbYHPSExHzJMc\\nHSfi2iy6YuYgLZFHgt0QVesq6h8aYN4D9EXrfXDO2Ndw5xKNjG2g17odMsU8UIJ+\\n1BuIF9GgZOTNURWDgSAC2MVqcp5U5QYEEOCjO9GppE9dCOj8\\\/qGpyiQqzbE5tStx\\nbh4CJXqppaa4Hs3tsulw3w==\"}",
            "remote_mozilla_sync": "mozilla_sync\/appinfo\/remote.php",
            "lastcron": "1512909903",
            "shareapi_default_expire_date": "yes",
            "shareapi_enforce_expire_date": "no",
            "shareapi_allow_mail_notification": "yes",
            "shareapi_only_share_with_group_members": "yes",
            "repairlegacystoragesdone": "yes",
            "shareapi_allow_public_notification": "yes",
            "backgroundjobs_mode": "cron",
            "shareapi_expire_after_n_days": "14",
            "remote_dav": "dav\/appinfo\/v2\/remote.php",
            "shareapi_exclude_groups": "no",
            "shareapi_exclude_groups_list": "[\"\"]",
            "vendor": "nextcloud",
            "updater.secret.created": "1512910726",
            "OC_Channel": "production",
            "moveavatarsdone": "yes",
            "previewsCleanedUp": "1",
            "umgmt_show_last_login": "false",
            "umgmt_show_backend": "false",
            "umgmt_show_email": "false",
            "umgmt_show_storage_location": "false",
            "installed.bundles": "[\"CoreBundle\"]",
            "scss.variables": "34937387f51adabf716957d984e03a04",
            "umgmt_send_email": "true",
            "oc.integritycheck.checker": "[]"
        },
        "dav": {
            "installed_version": "1.3.0",
            "types": "filesystem",
            "enabled": "yes",
            "OCA\\DAV\\Migration\\ValueFixInsert_ran": "true",
            "buildCalendarSearchIndex": "yes"
        },
        "deck": {
            "installed_version": "0.2.8",
            "enabled": "yes",
            "types": ""
        },
        "direct_menu": {
            "enabled": "no",
            "installed_version": "0.10.2",
            "types": "",
            "ocsid": "169148"
        },
        "documents": {
            "installed_version": "0.8.2",
            "types": "",
            "enabled": "no",
            "ocsid": "168711"
        },
        "drawio": {
            "installed_version": "0.8.8",
            "enabled": "yes",
            "types": "filesystem"
        },
        "external": {
            "installed_version": "1.2",
            "ocsid": "166046",
            "types": "",
            "enabled": "no"
        },
        "federatedfilesharing": {
            "installed_version": "1.2.0",
            "types": "",
            "enabled": "yes"
        },
        "federation": {
            "installed_version": "1.2.0",
            "types": "authentication",
            "enabled": "yes"
        },
        "files": {
            "installed_version": "1.7.2",
            "types": "filesystem",
            "enabled": "yes",
            "backgroundwatcher_previous_file": "734",
            "backgroundwatcher_previous_folder": "4701",
            "cronjob_scan_files": "500"
        },
        "files_accesscontrol": {
            "installed_version": "1.2.5",
            "types": "filesystem",
            "enabled": "yes"
        },
        "files_automatedtagging": {
            "installed_version": "1.2.2",
            "enabled": "yes",
            "types": "filesystem"
        },
        "files_downloadactivity": {
            "installed_version": "1.1.1",
            "enabled": "yes",
            "types": "filesystem"
        },
        "files_external": {
            "installed_version": "1.3.0",
            "ocsid": "166048",
            "types": "filesystem",
            "enabled": "yes",
            "user_mounting_backends": "dav,owncloud,sftp,amazons3,dropbox,googledrive,swift,smb,\\OC\\Files\\Storage\\SFTP_Key,\\OC\\Files\\Storage\\SMB_OC"
        },
        "files_locking": {
            "installed_version": "",
            "types": "filesystem",
            "enabled": "no"
        },
        "files_markdown": {
            "enabled": "yes",
            "installed_version": "2.0.1",
            "types": ""
        },
        "files_pdfviewer": {
            "installed_version": "1.1.1",
            "types": "",
            "enabled": "yes",
            "ocsid": "166049"
        },
        "files_retention": {
            "installed_version": "1.1.2",
            "types": "filesystem",
            "enabled": "yes"
        },
        "files_sharing": {
            "installed_version": "1.4.0",
            "types": "filesystem",
            "enabled": "yes",
            "lookupServerUploadEnabled": "no"
        },
        "files_texteditor": {
            "installed_version": "2.4.1",
            "types": "",
            "enabled": "yes",
            "ocsid": "166051"
        },
        "files_trashbin": {
            "installed_version": "1.2.0",
            "types": "filesystem",
            "enabled": "yes"
        },
        "files_versions": {
            "installed_version": "1.5.0",
            "types": "filesystem",
            "enabled": "yes"
        },
        "files_videoplayer": {
            "installed_version": "1.1.0",
            "types": "",
            "enabled": "yes"
        },
        "files_videoviewer": {
            "installed_version": "0.1.3",
            "types": "",
            "enabled": "no",
            "ocsid": "166054"
        },
        "firstrunwizard": {
            "installed_version": "2.1",
            "types": "logging",
            "enabled": "yes"
        },
        "gallery": {
            "installed_version": "17.0.0",
            "types": "",
            "enabled": "yes"
        },
        "gpxpod": {
            "enabled": "yes",
            "installed_version": "2.2.1",
            "types": ""
        },
        "logreader": {
            "installed_version": "2.0.0",
            "ocsid": "170871",
            "types": "",
            "enabled": "yes"
        },
        "lookup_server_connector": {
            "installed_version": "1.0.0",
            "types": "authentication",
            "enabled": "yes"
        },
        "mail": {
            "enabled": "no",
            "installed_version": "0.6.2",
            "types": ""
        },
        "metadata": {
            "installed_version": "0.6.0",
            "enabled": "yes",
            "types": ""
        },
        "mozilla_sync": {
            "installed_version": "1.4",
            "enabled": "no",
            "types": ""
        },
        "nextcloud_announcements": {
            "installed_version": "1.1",
            "types": "logging",
            "enabled": "yes",
            "pub_date": "Sat, 10 Dec 2016 00:00:00 +0100"
        },
        "notes": {
            "installed_version": "2.3.1",
            "types": "",
            "enabled": "yes",
            "ocsid": "174554"
        },
        "notifications": {
            "installed_version": "2.0.0",
            "types": "logging",
            "enabled": "yes"
        },
        "oauth2": {
            "installed_version": "1.0.5",
            "types": "authentication",
            "enabled": "yes"
        },
        "ocsms": {
            "enabled": "yes",
            "installed_version": "1.12.1",
            "types": "",
            "ocsid": "167289"
        },
        "ojsxc": {
            "enabled": "no",
            "installed_version": "3.1.1",
            "types": "prelogin",
            "serverType": "internal",
            "boshUrl": "https:\/\/***REMOVED SENSITIVE VALUE***\/http-bind\/",
            "xmppDomain": "***REMOVED SENSITIVE VALUE***",
            "xmppResource": "nextcloud",
            "xmppOverwrite": "true",
            "xmppStartMinimized": "false",
            "xmppPreferMail": "false",
            "iceUrl": "",
            "iceUsername": "",
            "iceCredential": "",
            "iceSecret": "",
            "iceTtl": "",
            "firefoxExtension": "",
            "chromeExtension": "",
            "externalServices": ""
        },
        "ownnote": {
            "installed_version": "1.07",
            "enabled": "no",
            "types": ""
        },
        "password_policy": {
            "installed_version": "1.2.2",
            "types": "",
            "enabled": "yes",
            "enforceNonCommonPassword": "0",
            "minLength": "3"
        },
        "polls": {
            "installed_version": "0.8.0",
            "enabled": "yes",
            "types": ""
        },
        "provisioning_api": {
            "installed_version": "1.2.0",
            "types": "prevent_group_restriction",
            "enabled": "yes"
        },
        "qownnotesapi": {
            "enabled": "no",
            "installed_version": "17.5.0",
            "types": "",
            "ocsid": "173817"
        },
        "quota_warning": {
            "installed_version": "1.1.1",
            "enabled": "yes",
            "types": "filesystem"
        },
        "search_lucene": {
            "installed_version": "0.5.3",
            "types": "filesystem",
            "enabled": "no",
            "ocsid": "168709"
        },
        "serverinfo": {
            "installed_version": "1.2.0",
            "types": "",
            "enabled": "yes"
        },
        "sharebymail": {
            "installed_version": "1.2.0",
            "types": "filesystem",
            "enabled": "yes",
            "sendpasswordmail": "no"
        },
        "spreed": {
            "enabled": "yes",
            "installed_version": "2.0.2",
            "types": "prevent_group_restriction"
        },
        "systemtags": {
            "installed_version": "1.2.0",
            "types": "logging",
            "enabled": "yes"
        },
        "tasks": {
            "installed_version": "0.9.5",
            "types": "",
            "enabled": "yes",
            "ocsid": "164356",
            "signed": "true"
        },
        "telephoneprovider": {
            "installed_version": "1.0.0",
            "enabled": "yes",
            "types": ""
        },
        "templateeditor": {
            "installed_version": "0.2",
            "types": "",
            "enabled": "no"
        },
        "theming": {
            "installed_version": "1.3.0",
            "types": "logging",
            "enabled": "yes",
            "backgroundMime": "image\/jpeg",
            "cachebuster": "1"
        },
        "twofactor_backupcodes": {
            "installed_version": "1.1.1",
            "types": "",
            "enabled": "yes"
        },
        "twofactor_totp": {
            "enabled": "yes",
            "installed_version": "1.3.1",
            "types": ""
        },
        "updatenotification": {
            "installed_version": "1.2.0",
            "types": "",
            "enabled": "yes",
            "contacts": "2.0.1",
            "core": "13.0.0.6",
            "tasks": "0.9.5",
            "calendar": "1.5.7",
            "qownnotesapi": "17.5.0",
            "gpxpod": "2.2.1",
            "direct_menu": "0.10.2",
            "ocsms": "1.12.1",
            "bookmarks": "0.10.1",
            "notes": "2.3.2",
            "update_check_errors": "0",
            "twofactor_totp": "1.3.1",
            "quota_warning": "1.1.1",
            "files_markdown": "2.0.1",
            "files_accesscontrol": "1.2.5",
            "deck": "0.2.8"
        },
        "updater": {
            "installed_version": "0.4",
            "types": "",
            "enabled": "no",
            "ocsid": "166059"
        },
        "user_external": {
            "installed_version": "0.4",
            "ocsid": "166060",
            "types": "authentication,prelogin",
            "enabled": "yes"
        },
        "workflowengine": {
            "installed_version": "1.2.0",
            "types": "filesystem",
            "enabled": "yes"
        }
    }
}
```
</details>

**Are you using external storage, if yes which one:** No

**Are you using encryption:** No

**Are you using an external user-backend, if yes which one:** No

**Comment by masaki-iwata** [comment]
I too suffering from the same error.
Upgrad from NC12.0.4 to NC13 beta3.

Operating system: CentOS 6.2
Web server: Apache 2.2
PHP version: 7.1.2

**Comment by MorrisJobke** [comment]
> - files_accesscontrol: 1.2.5

This was fixed in files_accesscontrol 1.3.0 - https://github.com/nextcloud/files_accesscontrol/pull/75

**Comment by Homer-Sim** [comment]
Hello and sorry ofr picking it up again.

I tried to update today from 12.0.3 to 13.0.1 and receiving right this error message.
What was the solution for this issue?

Thanks
